samples/net/sockets/echo_server/CMakeLists.txt

```diff
@@ -42,3 +42,7 @@ foreach(inc_file
     ${gen_dir}/${inc_file}.inc
     )
 endforeach()
+
+if(CONFIG_NET_CONFIG_USE_YAML)
+  network_generate_config_file_for_target(app ${CONFIG_NET_CONFIG_YAML_FILE})
```

Comment:
Is it needed to generate config file in the sample's `CMakeLists.txt`, or could it be done in a more generic way in config's `CMakeLists.txt` (it's needed to enable the feature with Kconfig anyway).

Reply:
IMHO the configuration yaml file should not be placed to `subsys/net/lib/config` directory, but near where it is needed. It could be in one directory too if we want to share these with tests/samples.

Reply:
I tried to place these settings to config lib CMakefilẹ file. Unfortunately the system cannot find the generated file in that case.
```
ninja: error: '..../zephyr/subsys/net/lib/config/configuration.yaml', needed by 'zephyr/include/generated/net_init_config.inc', missing and no known rule to make it
```

Reply:
I agree with @rlubos that we should investigate further how to apply the "convention over configuration" approach here.